universal-login-sdk/test/core/observers/ObserverRunner.test.ts

```diff
@@ -0,0 +1,40 @@
+import {expect} from 'chai';
+import {sleep} from '@unilogin/commons';
+import {ObserverRunnerUnderTest} from './ObserverRunnerUnderTest';
+
+describe('UNIT: ObserverRunner', () => {
+  const tick = 1000;
+  let observerRunner: ObserverRunnerUnderTest;
+
+  beforeEach(() => {
+    observerRunner = new ObserverRunnerUnderTest(tick);
+  });
+
+  it('finalizeAndStop waits for execute to be done with await', async () => {
+    expect(observerRunner.iterator).to.eq(0);
+    observerRunner.start();
+    expect(observerRunner.iterator).to.eq(1);
+    await observerRunner.finalizeAndStop();
+    expect(observerRunner.iterator).to.eq(2);
+  });
+
+  it('finalizeAndStop waits for execute to be done without await', async () => {
+    expect(observerRunner.iterator).to.eq(0);
+    observerRunner.start();
+    expect(observerRunner.iterator).to.eq(1);
+    const stopPromise = observerRunner.finalizeAndStop();
+    expect(observerRunner.iterator).to.eq(1);
+    await stopPromise;
+    expect(observerRunner.iterator).to.eq(2);
+  });
+
+  it('stop doesn`t wait for execute to be finished', async () => {
+    expect(observerRunner.iterator).to.eq(0);
+    observerRunner.start();
+    expect(observerRunner.iterator).to.eq(1);
+    observerRunner.stop();
+    expect(observerRunner.iterator).to.eq(1);
+    await sleep(tick);
+    expect(observerRunner.iterator).to.eq(2);
+  });
+});
```

Comment:
I would add a test that waits another tick and sees that it still is two. And another test in which the loop executes multiple times.